### PR TITLE
Change integer Enums to IntEnum

### LINKED ISSUE #353: Migrate Enum to IntEnum
On python 3.6 a new enum type was introduced, `IntEnum`. This is used when the enumerations are matched against integers. From this point on, traditional Enums are not going to match against integers anymore. So, we are not going to be able to match some of our enums against integers, for example:

Current **pyof/v0x04/common/header.py**:
```python
class Type(Enum):
    """Enumeration of Message Types."""

    # Symetric/Immutable messages
    OFPT_HELLO = 0
    OFPT_ERROR = 1
    OFPT_ECHO_REQUEST = 2
    OFPT_ECHO_REPLY = 3
    OFPT_EXPERIMENTER = 4
    ...
```
```python
In [1]: from pyof.v0x04.common.header import Type
In [2]: Type.OFPT_HELLO == 0
Out[2]: False
```

But, if we change it to use IntEnum:

Proposed **pyof/v0x04/common/header.py**:
```python
class Type(IntEnum):
    """Enumeration of Message Types."""

    # Symetric/Immutable messages
    OFPT_HELLO = 0
    OFPT_ERROR = 1
    OFPT_ECHO_REQUEST = 2
    OFPT_ECHO_REPLY = 3
    OFPT_EXPERIMENTER = 4
    ...
```
```python
In [1]: from pyof.v0x04.common.header import Type
In [2]: Type.OFPT_HELLO == 0
Out[2]: True
```

Off course that this wouldn't apply to all enumerations (since we do have some that use binary/bytes representations), but we would use it on all that only use integers.

Currently our tests here are not failing due to this error.

### PR DESCRIPTION
**Pull request by renanrodrigo**
Fix #353 
Also small refactoring to validations, given that `isinstance(some_enum_attribute, int) == True` now.

### CODE REVIEW

File: pyof/foundation/base.py [hunk 1]
```diff
@@ -224,7 +224,8 @@ def isenum(self):
         Returns:
             bool: Whether it is an :class:`~Enum`.
         """
-        return self.enum_ref and issubclass(self.enum_ref, Enum)
+        return self.enum_ref and (issubclass(self.enum_ref, Enum) or
```
Comment (diraol) [inline pyof/foundation/base.py]: @renanrodrigo, please, change here to:
```return self.enum_ref and issubclass(self.enum_ref, (Enum, IntEnum))```

Comment (renanrodrigo) [inline pyof/foundation/base.py]: done

**Comment by diraol** [comment]
@renanrodrigo please, update your branch agains the master ;)

**Comment by beraldoleal** [comment]
@renanrodrigo scrutinizer tests, please!

**Comment by renanrodrigo** [comment]
They are happening because of the 'new' linter errors. Will fix that.

**Comment by renanrodrigo** [comment]
Now rebased and passing (:

**Pull request merged**